File: p1/ejercicio3/ataque.py
from abc import ABC, abstractmethod


class Ataque(ABC):
    desgaste = 0
    @abstractmethod
    def atacar(self):
        pass

    @abstractmethod
    def danio(self):
        pass

class AtaqueAereo(Ataque):
    desgaste = 10
    
    def atacar(self):
        print("Ataque aereo")

    def danio(self):
        return self.desgaste
        

class AtaqueTerrestre(Ataque):
    desgaste = 5
    
    def atacar(self):
        print("Ataque aereo")

    def danio(self):
        return self.desgaste
        

class AtaqueMaritimo(Ataque):
    desgaste = 20
    
    def atacar(self):
        print("Ataque aereo")

    def danio(self):
        return self.desgaste

File: pr2/militares/lib/ataque.dart
import "militar.dart";

abstract class Ataque{
  double danio = 0;


  void atacar(Militar m){
    m.recibeAtaque(danio);
  }

  String totxt();
}

class AtaqueAereo extends Ataque{
  AtaqueAereo(){
    danio=5;
  }
  
  @override
  String totxt(){
    return "Ataque Aereo";
  }
}

@override
class AtaqueTerrrestre extends Ataque{
  AtaqueTerrrestre(){
    danio=1;
  }

  @override
  String totxt(){
    return "Ataque Terrestre";
  }
}

class AtaqueMaritimo extends Ataque{
  AtaqueMaritimo(){
    danio=2;
  }

  @override
  String totxt(){
    return "Ataque Maritimo";
  }
  
}
